MUSL build fails on OSX

Hello, 

I have been trying `cargo build` on  `rust-htslib`:
```
$ git clone --recursive https://github.com/rust-bio/rust-htslib.git
$ cd rust-htslib 
$ cargo build
```
which output a whole lot warning on the c source code and a fatal
```
    error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/libz-sys-7c94d09687ec3ebf/out/include" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/bzip2-sys-1b34115e67e42ca9/out/include" "-I" "/Users/admin/.cargo/registry/src/github.com-1ecc6299db9ec823/lzma-sys-0.1.16/xz-5.2/src/liblzma/api" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/curl-sys-6886d1c85d5521de/out/include" "-Wall" "-Wextra" "-o" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.o" "-c" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.c" with args "cc" did not execute successfully (status code exit code: 1).
```

I have also tried
```
$ brew install FiloSottile/musl-cross/musl-cross
$ brew install bzip2 zlib xz curl-openssl
$ cross build
```
with the same result. 

However, `cross build --target x86_64-unknown-linux-musl` does work. 

Other info: 
* OS: MacOS Catalina 10.15.6 
* 
```
$ cc --version
Apple clang version 12.0.0 (clang-1200.0.31.1)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

(Sorry for the spam) 
Running the failed `cc` command manually in verbose mode generate the following (warnings removed):
```
cc -v "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/libz-sys-7c94d09687ec3ebf/out/include" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/bzip2-sys-1b34115e67e42ca9/out/include" "-I" "/Users/admin/.cargo/registry/src/github.com-1ecc6299db9ec823/lzma-sys-0.1.16/xz-5.2/src/liblzma/api" "-I" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/curl-sys-6886d1c85d5521de/out/include" "-Wall" "-Wextra" "-o" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.o" "-c" "/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.c"
Apple clang version 12.0.0 (clang-1200.0.31.1)
Target: x86_64-apple-darwin19.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
 "/Library/Developer/CommandLineTools/usr/bin/clang" -cc1 -triple x86_64-apple-macosx10.15.0 -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -discard-value-names -main-file-name sam.c -mrelocation-model pic -pic-level 2 -mthread-model posix -mframe-pointer=all -fno-strict-return -masm-verbose -munwind-tables -target-sdk-version=10.15.6 -fcompatibility-qualified-id-block-type-checking -target-cpu penryn -dwarf-column-info -debug-info-kind=standalone -dwarf-version=4 -debugger-tuning=lldb -target-linker-version 609 -v -ffunction-sections -fdata-sections -resource-dir /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -I /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib -I /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/libz-sys-7c94d09687ec3ebf/out/include -I /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/bzip2-sys-1b34115e67e42ca9/out/include -I /Users/admin/.cargo/registry/src/github.com-1ecc6299db9ec823/lzma-sys-0.1.16/xz-5.2/src/liblzma/api -I /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/curl-sys-6886d1c85d5521de/out/include -I/usr/local/include -internal-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/local/include -internal-isystem /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/include -internal-externc-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include -internal-externc-isystem /Library/Developer/CommandLineTools/usr/include -O0 -Wall -Wextra -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -fdebug-compilation-dir /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib -ferror-limit 19 -fmessage-length 318 -stack-protector 1 -fstack-check -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fobjc-runtime=macosx-10.15.0 -fmax-type-align=16 -fdiagnostics-show-option -fcolor-diagnostics -o /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.o -x c /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.c
clang -cc1 version 12.0.0 (clang-1200.0.31.1) default target x86_64-apple-darwin19.6.0
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/local/include"
ignoring nonexistent directory "/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib
 /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/libz-sys-7c94d09687ec3ebf/out/include
 /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/bzip2-sys-1b34115e67e42ca9/out/include
 /Users/admin/.cargo/registry/src/github.com-1ecc6299db9ec823/lzma-sys-0.1.16/xz-5.2/src/liblzma/api
 /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/curl-sys-6886d1c85d5521de/out/include
 /usr/local/include
 /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include
 /Library/Developer/CommandLineTools/usr/include
 /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks (framework directory)
End of search list.
In file included from /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.c:46:
In file included from /Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/cram/cram.h:52:
/Users/admin/Desktop/MGHA_bioinformatics/PROJECT_rust_bio/rust-htslib/target/debug/build/hts-sys-bf3eaa6abc104e1b/out/htslib/sam.c:2331:21: error: implicit declaration of function 'usleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                    usleep(10000);
                    ^
47 warnings and 1 error generated.
```

```

More errors are cropping up with OSX, tricky target without cross/docker on the GitHub Actions runner, i.e:

```
% cargo build --all-features --target x86_64-unknown-linux-musl
   Compiling hts-sys v1.11.0 (/Users/rvalls/dev/umccr/htsget/rust-htslib/hts-sys)
The following warnings were emitted during compilation:

warning: htslib/hfile_s3.c:67:37: fatal error: CommonCrypto/CommonHMAC.h: No such file or directory
warning:  #include <CommonCrypto/CommonHMAC.h>
warning:                                      ^
warning: compilation terminated.

error: failed to run custom build command for `hts-sys v1.11.0 (/Users/rvalls/dev/umccr/htsget/rust-htslib/hts-sys)`
```

Which seems to be related to this particular rabbit hole: https://github.com/onmyway133/Arcane/issues/4

> More errors are cropping up with OSX, tricky target without cross/docker on the GitHub Actions runner, i.e:
> 
> ```
> % cargo build --all-features --target x86_64-unknown-linux-musl
>    Compiling hts-sys v1.11.0 (/Users/rvalls/dev/umccr/htsget/rust-htslib/hts-sys)
> The following warnings were emitted during compilation:
> 
> warning: htslib/hfile_s3.c:67:37: fatal error: CommonCrypto/CommonHMAC.h: No such file or directory
> warning:  #include <CommonCrypto/CommonHMAC.h>
> warning:                                      ^
> warning: compilation terminated.
> 
> error: failed to run custom build command for `hts-sys v1.11.0 (/Users/rvalls/dev/umccr/htsget/rust-htslib/hts-sys)`
> ```
> 
> Which seems to be related to this particular rabbit hole: [onmyway133/Arcane#4](https://github.com/onmyway133/Arcane/issues/4)

This only seems to affect MUSL builds on OSX (which has limited usecases). Native (x86_64) seem to work fine. It's a pity since this affects AWS lambda deployments from an OSX build environment :/

--- a/hts-sys/build.rs
+++ b/hts-sys/build.rs
@@ -66,6 +66,10 @@ fn main() {
     }
     copy_directory("htslib", &out).unwrap();
 
+    // In build.rs cfg(target_os) does not give the target when cross-compiling;
+    // instead use the environment variable supplied by cargo, which does.
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+
     let mut cfg = cc::Build::new();
 
     // default files
@@ -137,14 +141,10 @@ fn main() {
             cfg.file("htslib/hfile_libcurl.c");
             println!("cargo:rerun-if-changed=htslib/hfile_libcurl.c");
 
-            #[cfg(target_os = "macos")]
-            {
+            if target_os == "macos" {
                 // Use builtin MacOS CommonCrypto HMAC
                 config_lines.push("#define HAVE_COMMONCRYPTO 1");
-            }
-
-            #[cfg(not(target_os = "macos"))]
-            {
+            } else {
                 if let Ok(inc) = env::var("DEP_OPENSSL_INCLUDE").map(PathBuf::from) {
                     // Must use hmac from libcrypto in openssl
                     cfg.include(inc);
@@ -209,15 +209,12 @@ fn main() {
     }
 
     // If no bindgen, use pre-built bindings
-    #[cfg(all(not(feature = "bindgen"), target_os = "macos"))]
-    {
+    #[cfg(not(feature = "bindgen"))]
+    if target_os == "macos" {
         fs::copy("osx_prebuilt_bindings.rs", out.join("bindings.rs"))
             .expect("couldn't copy prebuilt bindings");
         println!("cargo:rerun-if-changed=osx_prebuilt_bindings.rs");
-    }
-
-    #[cfg(all(not(feature = "bindgen"), target_os = "linux"))]
-    {
+    } else {
         fs::copy("linux_prebuilt_bindings.rs", out.join("bindings.rs"))
             .expect("couldn't copy prebuilt bindings");
         println!("cargo:rerun-if-changed=linux_prebuilt_bindings.rs");

